.github/workflows/pr.yml

```diff
@@ -0,0 +1,154 @@
+name: PR Build
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+env:
+  REALM_CI: true
+
+jobs:
+  baas-linux:
+    runs-on: ubuntu-latest
+    name: BaaS Linux
+    outputs:
+      clusterName: ${{ steps.deploy-mdb-apps.outputs.clusterName }}
+    timeout-minutes: 20
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          submodules: false
+      - uses: realm/ci-actions/mdb-realm/deployApps@fac1d6958f03d71de743305ce3ab27594efbe7b7
+        id: deploy-mdb-apps
+        with:
+          projectId: ${{ secrets.ATLAS_PROJECT_ID }}
+          apiKey: ${{ secrets.ATLAS_PUBLIC_API_KEY }}
+          privateApiKey: ${{ secrets.ATLAS_PRIVATE_API_KEY }}
+          differentiator: dart-linux
+      - name : Setup Dart SDK
+        uses: dart-lang/setup-dart@main
+        with:
+          sdk: stable
+      - name: Deploy Apps
+        run: dart run realm_dart deploy-apps --baas-url https://realm-dev.mongodb.com --atlas-cluster ${{ steps.deploy-mdb-apps.outputs.clusterName }} --api-key ${{ secrets.ATLAS_PUBLIC_API_KEY }} --private-api-key ${{ secrets.ATLAS_PRIVATE_API_KEY }} --project-id ${{ secrets.ATLAS_PROJECT_ID }}
+  build-linux:
+    runs-on: ubuntu-latest
+    name: Build Linux
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          submodules: 'recursive'
+      - name: Setup Ninja
+        uses: seanmiddleditch/gha-setup-ninja@master
+      - name: Build Realm Dart for Linux
+        run: ./scripts/build-linux.sh
+      - name: Store artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: librealm-linux
```

Comment:
You need to name this after a unique finger-print of the sources. Previously I used `git submodule status` on `src/realm-core`, but that was not deemed good enough, since it doesn't capture changes to the dart specific files in `src`. Hence I suggest also adding `sha1sum src/* | sha1sum` as well. Also, I suggest you use `actions/cache@v3` instead.

Reply:
This is different from caching as it creates an artifact for this particular run. Artifacts are not shared between runs, so they can have the same name. The general recommendation for passing build outputs from one job another is to use artifacts as it has several advantages:
1. They can not be accidentally overwritten by other runs.
2. They can be downloaded after the run completes (they appear as files in the workflow summary).
3. They can be downloaded deterministically from other workflows - e.g. once we have a proper release workflow, we can find the run for the correct commit and download the artifacts from it.

Reply:
But I would like to re-use it another run, so that I can avoid building `librealm-dart` unless there are changes in `src`. We can also upload it as an artifact, to satisfy 2 and 3.

Reply:
So what about building and caching, but still use an artifact to pass output to next job.

Reply:
Yes, I do plan to add ccache to speed up builds, but that's out of scope for what this PR tries to achieve. In the .NET workflows, we do use a combination of caching and artifact uploads to avoid rebuilding the native binaries and we'll bring some of these mechanics to dart, but I'd rather do it step by step rather than as a single massive PR that'd be a lot harder to review.

Reply:
Okay. Point taken 👍